cpp/test/sg/rf_treelite_test.cu

```diff
@@ -146,24 +146,19 @@ class RfTreeliteTestCommon : public ::testing::TestWithParam<RfInputs<T>> {
             params.n_cols, predicted_labels_d, false);
     CUDA_CHECK(cudaStreamSynchronize(stream));
 
-    if (is_classification) {
-      predicted_labels_h.resize(params.n_inference_rows);
-      CUDA_CHECK(cudaMemcpy(predicted_labels_h.data(), predicted_labels_d,
-                            sizeof(L) * params.n_inference_rows,
-                            cudaMemcpyDeviceToHost));
-    } else {
-      // In case of regression, predicted_labels_d alreay has the final result.
-      ref_predicted_labels.resize(params.n_inference_rows);
-      CUDA_CHECK(cudaMemcpy(ref_predicted_labels.data(), predicted_labels_d,
-                            sizeof(L) * params.n_inference_rows,
-                            cudaMemcpyDeviceToHost));
-    }
+    predicted_labels_h.resize(params.n_inference_rows);
+    ref_predicted_labels.resize(params.n_inference_rows);
+
+    updateHost(predicted_labels_h.data(), predicted_labels_d,
```

Comment:
`cudaStreamSynchronize(stream) is necessary after `updateHost()` before the first use of the host array.